src/composed/message/decrypt.rs

```diff
@@ -1,62 +1,26 @@
 use crate::crypto::sym::SymmetricKeyAlgorithm;
-use crate::crypto::{checksum, ecdh, rsa};
 use crate::errors::Result;
 use crate::packet::SymKeyEncryptedSessionKey;
 use crate::types::{KeyTrait, Mpi, SecretKeyRepr, SecretKeyTrait};
 
 /// Decrypts session key using secret key.
-pub fn decrypt_session_key<F>(
-    locked_key: &(impl SecretKeyTrait + KeyTrait),
+pub fn decrypt_session_key<F, L>(
+    locked_key: &L,
     key_pw: F,
     mpis: &[Mpi],
 ) -> Result<(Vec<u8>, SymmetricKeyAlgorithm)>
 where
     F: FnOnce() -> String,
+    L: SecretKeyTrait<Unlocked = SecretKeyRepr> + KeyTrait,
```

Comment:
I'm using `decrypt_session_key` in here: https://codeberg.org/wiktor/openpgp-card-rpgp/src/branch/hack/examples/decrypt.rs#L27

But the `Unlocked` needing a `SecretKeyRepr` constraint here means the `decrypt_session_key` is still bound to software keys... I know this is a work in progress but maybe I'm holding it wrong somehow?

Reply:
you are not holding it wrong, you can't use this yet for other implementations, I just added this as a placeholder until I have a constraint that I am happy with

Reply:
Ack. Thanks! :+1: 

I think in the ideal world the `SecretKeyRepr` would never be passed anywhere but some trait which `SecretKeyRepr` (or the Secret key packet?) implements would be used everywhere.

From what I can see there's just a couple of bits needed to pass around:
  - ECDH key params (even though in v6 they'd be fixed),
  - some metadata e.g. for signing one embeds the signing key fingerprint in the signature